libvast/src/system/explorer.cpp

```diff
@@ -0,0 +1,160 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#include "vast/system/explorer.hpp"
+
+#include "vast/command.hpp"
+#include "vast/concept/printable/to_string.hpp"
+#include "vast/concept/printable/vast/expression.hpp"
+#include "vast/defaults.hpp"
+#include "vast/detail/string.hpp"
+#include "vast/expression.hpp"
+#include "vast/logger.hpp"
+#include "vast/system/atoms.hpp"
+#include "vast/system/exporter.hpp"
+#include "vast/table_slice.hpp"
+#include "vast/table_slice_builder.hpp"
+#include "vast/table_slice_builder_factory.hpp"
+
+#include <caf/event_based_actor.hpp>
+
+#include <algorithm>
+
+namespace vast::system {
+
+explorer_state::explorer_state(caf::event_based_actor*) {
+  // nop
+}
+
+void explorer_state::forward_results(vast::table_slice_ptr slice) {
```

Comment:
For the future reader/maintainer, this function could use some comments for every step.